pkg/controller/chaosengine/chaosengine_controller_test.go

```diff
@@ -47,6 +50,11 @@ var (
 	clientSet  *chaosClient.LitmuschaosV1alpha1Client
 )
 
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler1(mgr manager.Manager) *ReconcileChaosEngine {
+	return &ReconcileChaosEngine{client: mgr.GetClient(), scheme: mgr.GetScheme(), recorder: mgr.GetEventRecorderFor("chaos-operator")}
+}
+
 func init() {
 	kubeconfig := os.Getenv("HOME") + "/.kube/config"
```

Comment:
So, this would only work if you have a cluster available, right?
But this should not be the case, unit tests should be independent of hard-codings.
So, just take a look at how to initilze a https://godoc.org/k8s.io/client-go/kubernetes/fake
Ref: https://github.com/openebs/maya/blob/a61c18e3eeabc18befedb3e9890e5c6ef292e912/pkg/webhook/webhook_test.go#L28

Reply:
Everything else seems good, just check how test performs in these cases.

Reply:
Thanks @rahulchheda It is really helpful

Reply:
This will be refactored - tracked via https://github.com/litmuschaos/chaos-operator/issues/226